.github/workflows/build-and-push-image-build.yml

```diff
@@ -37,17 +37,10 @@ jobs:
   model-build-and-push-image:
     needs:
       - base-spack-build-and-push-image
-    if: needs.base-spack-build-and-push-image.result == 'success' || needs.base-spack-build-and-push-image.result == 'skipped'
+    if: always()
```

Comment:
Do we need an `if` at all if we're always running it?

Reply:
Answered my own question: this is required if a previous step might error and you still want this to run

see https://stackoverflow.com/a/58859404

Reply:
Yeah this is for the case where we skip the above base-spack-build-and-push-image (because that base-spack image already exists). We still need it to have pushed the image before we do any of the other dependency image stuff, though.